### PR TITLE
Added post-main e2e with lifecycle manager job

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added post-main e2e with lifecycle manager job

**Related issue(s)**

- https://github.com/kyma-project/nats-manager/issues/76

### CODE REVIEW

File: .github/workflows/push-with-lifecycle-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+      - name: Deploy lifecycle-manager
+        run: |
+          make -C hack/ci/ install-lifecycle-manager
+
+      - name: Wait for the 'post-nats-manager-build' job to succeed
+        uses: mfaizanse/wait-for-git-commit-status-action@v1.0.6
+        with:
+          context: "post-nats-manager-build"
+          commit_ref: "${{ github.sha }}"
```
Comment (mfaizanse) [inline .github/workflows/push-with-lifecycle-manager.yaml:41]:
```
context: "post-nats-manager-build"
commit_ref: "${{ github.sha }}"
```
This is different from pull-job.

File: .github/workflows/push-with-lifecycle-manager.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          GITHUB_OWNER: "${{ github.repository_owner }}"
+          GITHUB_REPO: "nats-manager"
+
+      - name: Wait for the 'post-nats-module-build' job to succeed
+        id: wait-nats-module-build
+        uses: mfaizanse/wait-for-git-commit-status-action@v1.0.6
+        with:
+          context: "post-nats-module-build"
+          commit_ref: "${{ github.sha }}"
```
Comment (mfaizanse) [inline .github/workflows/push-with-lifecycle-manager.yaml:55]:
```
context: "post-nats-manager-build"
commit_ref: "${{ github.sha }}"
```
This is different from pull-job.

File: .github/workflows/push-with-lifecycle-manager.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+          # The check interval is kept long otherwise it will exhaust the GitHub rate limit (More info: https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting)
+          check_interval: 60000 # 1 minute in milliseconds
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+          GITHUB_OWNER: "${{ github.repository_owner }}"
+          GITHUB_REPO: "nats-manager"
+
+      - name: Download & validate NATS module template
+        env:
+          COMMIT_STATUS_JSON: "${{ steps.wait-nats-module-build.outputs.json }}"
```
Comment (mfaizanse) [inline .github/workflows/push-with-lifecycle-manager.yaml:66]:
```
PR_NUMBER is removed.
```
This is different from pull-job.

File: .github/workflows/push-with-lifecycle-manager.yaml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install k3d tools
+        uses: nolar/setup-k3d-k3s@v1
+        with:
+          # This is the Kubernetes version used by k3d. See https://kubernetes.io/releases/
+          version: v1.26.5
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply .github/workflows/push-with-lifecycle-manager.yaml:23]: According to [docs](https://github.com/nolar/setup-k3d-k3s#version), it is a required field.

[the remark this answers is not in the document]
Comment (friedrichwilken) [reply .github/workflows/push-with-lifecycle-manager.yaml:23]: strange, but okay.

**Pull request merged**